expose pitchfork worker in env

ollym opened the pull request:
which is helpful for logging for example worker.nr in app error handling. we're in the process of tracking down a SSL session corruption bug in our HTTP library and we're trying to work out when it occurs if its happening across all workers or just one, it's impossible to tell at the moment because the hostname for all the errors is the same, being about to log `worker.nr` alongside that would answer the question

casperisfine commented:
> we're in the process of tracking down a SSL session corruption bug in our HTTP library

Out of curiosity, what does this look like?

As you likely know, all connections should be either closed before fork, or abandoned after fork. So perhaps your HTTP library keeps one open?

casperisfine commented:
Thanks for the link.

`OpenSSL` has a number of native locks, and if you fork while a background thread is holding one of these locks it can indeed lead to a broken process. Do you have any background thread doing things?

It could also be unrelated to your HTTP library, and be another SSL enabled network client (redis, database, etc) that is doing things in the background while you refork.

ollym commented:
So the reason we picked httpx over others is that it specifically didn't use threads to achieve concurrency and instead used native (Ruby) HTTP/2 multiplexing falling back to HTTP/1.1 pipelining. We then also intentionally removed all background threads within our app for other reasons, so there should be no background threads anywhere other than some specific gems that maintain thread pools eg. Bugsnag or SemanticLogger. We use HTTPX heavily, like 15,000 requests a minute.

We've kept all the rest of our stack very simple, we switched to solid_cache/good_job and removed our Redis dependency, and our postgres instance doesn't use SSL.

We haven't seen this SSL issue in longer than we'd expect since we rolled out the hotfix mentioned in that issue link, so we expect @HoneyryderChuck figured it out in the PR but we're waiting until the end of the week to be sure.
